solr/core/src/test/org/apache/solr/handler/admin/api/ReplaceNodeAPITest.java

```diff
@@ -0,0 +1,85 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.handler.admin.api;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Optional;
+import org.apache.solr.SolrTestCaseJ4;
+import org.apache.solr.cloud.OverseerSolrResponse;
+import org.apache.solr.cloud.api.collections.DistributedCollectionConfigSetCommandRunner;
+import org.apache.solr.common.util.NamedList;
+import org.apache.solr.core.CoreContainer;
+import org.apache.solr.handler.admin.CollectionsHandler;
+import org.apache.solr.jersey.SolrJerseyResponse;
+import org.apache.solr.request.SolrQueryRequest;
+import org.apache.solr.response.SolrQueryResponse;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/** Unit tests for {@link ReplaceNodeAPI} */
+public class ReplaceNodeAPITest extends SolrTestCaseJ4 {
+
+  private CoreContainer mockCoreContainer;
+  private static final String sourceNodeName = "demoSourceNode";
+  private static final String targetNodeName = "demoTargetNode";
+  public static final String async = "async";
+  private static final boolean waitForFinalState = false;
+  private SolrQueryRequest mockQueryRequest;
+  private SolrQueryResponse queryResponse;
+  private ReplaceNodeAPI replaceNodeApi;
+  private DistributedCollectionConfigSetCommandRunner mockCommandRunner;
+
+  @BeforeClass
+  public static void ensureWorkingMockito() {
+    assumeWorkingMockito();
+  }
+
+  @Override
+  @Before
+  public void setUp() throws Exception {
+    super.setUp();
+
+    mockCoreContainer = mock(CoreContainer.class);
+    mockCommandRunner = mock(DistributedCollectionConfigSetCommandRunner.class);
+    when(mockCoreContainer.getDistributedCollectionCommandRunner())
+        .thenReturn(Optional.of(mockCommandRunner));
+    when(mockCommandRunner.runCollectionCommand(any(), any(), anyLong()))
+        .thenReturn(new OverseerSolrResponse(new NamedList<>()));
+    mockQueryRequest = mock(SolrQueryRequest.class);
+    queryResponse = new SolrQueryResponse();
+    replaceNodeApi = new ReplaceNodeAPI(mockCoreContainer, mockQueryRequest, queryResponse);
+  }
+
+  @Test
+  public void testSuccessfulReplaceNodeCommand() throws Exception {
```

Comment:
[-1] I think we should either re-work this test a bit to assert on something more meaningful (e.g. the remote-message that gets created from all the API parameters), or we should lean exclusively on the JerseyTest you mentioned in some comments.

Either approach (or some third option if you've got other ideas) is totally fine with me.  But as-is at least this test just doesn't validate all that much.

If you want to go with the first option (reworking this to assert on the remote-message), then AddReplicaPropertyAPITest might be a good reference.  That test makes an API call and uses a special type of mock called an "ArgumentCaptor" to record the "remote message", so that the test can assert on it later.

The key bits in AddReplicaPropertyAPITest are:

```
    messageCapturer = ArgumentCaptor.forClass(ZkNodeProps.class);
```
(creates the argument-captor)

```
    addReplicaPropApi.addReplicaProperty(
        "someColl", "someShard", "someReplica", "somePropName", ANY_REQ_BODY);
    verify(mockCommandRunner).runCollectionCommand(messageCapturer.capture(), any(), anyLong());
```
(calls the new API and "captures" or records the remote-message)